From 69fb11209fdea0ed23d8868a853f80b9c931d090 Mon Sep 17 00:00:00 2001
From: "guangxi.Chen" <gxchen@nd.com.cn>
Date: Wed, 16 Aug 2023 19:05:07 +0800
Subject: [PATCH] AP9-22472 QA Only - Zoom.us didn't recognize USB camera in
 Guest Account but did when signed into Promethean Account

fix version:REG_AP9_2.0.0

why:For websites in chromium such as zoom.us, due to the variable transfer between pages, if the number of camera lists obtained on the home page is 0, subsequent refreshes of other pages or the current page will not continue to refresh the current number of cameras, resulting in subsequent Camera plugging and unplugging cannot be recognized by correct refresh.
---
 .../java/src/org/chromium/media/VideoCaptureCamera2.java        | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/media/capture/video/android/java/src/org/chromium/media/VideoCaptureCamera2.java b/media/capture/video/android/java/src/org/chromium/media/VideoCaptureCamera2.java
index 412f5b38dc1b8..70e40536e97c5 100644
--- a/media/capture/video/android/java/src/org/chromium/media/VideoCaptureCamera2.java
+++ b/media/capture/video/android/java/src/org/chromium/media/VideoCaptureCamera2.java
@@ -1360,7 +1360,7 @@ public class VideoCaptureCamera2 extends VideoCapture {
         }
         if (manager == null) return 0;
         try {
-            return manager.getCameraIdList().length;
+            return Math.max(1, manager.getCameraIdList().length);
         } catch (CameraAccessException | SecurityException | AssertionError ex) {
             // SecurityException is undocumented but seen in the wild: https://crbug/605424.
             Log.e(TAG, "getNumberOfCameras: getCameraIdList(): ", ex);
-- 
2.34.1

